### ISSUE TITLE
Mission generation should use NAV_DELAY instead of sending two waypoints on the same location

### ISSUE THREAD

**Issue by LorenzMeier** [mavlink/MAVSDK#1101]
From @julianoes:
> The reason MAVSDK sends the same lat/lon twice in a row is because it adds a LOITER_TIME right after a NAV_WAYPOINT. The right approach would be to use NAV_DELAY [...].

Which is implemented in PX4 for quite some time (I verified). The reason PX4 rejects now waypoints that are on top of each other is because the path between those waypoints is not defined (zero distance). Often we however need to know where the vehicle is coming from geometrically and allowing N waypoints on the same location would force us to traverse the mission multiple times and quite a bit forward and backward, increasing the complexity of the autopilot in terms of memory, execution time and also in terms of our ability to QA and certify the code.

So in short: This is undesirable in principle and navigation commands should not exist in the same physical location one after the other.

**Comment by julianoes** [comment]
> So in short: This is undesirable in principle and navigation commands should not exist in the same physical location one after the other.

If this is a hard rule it should also be added to the MAVLink standard.

**Comment by dayjaby** [comment]
There are a couple of scenarios that cannot be done if there is such a hard rule:
(PREVIOUS WAYPOINT), WAYPOINT, DELAY, IMAGE_START_CAPTURE, LOITER_UNLIMITED for example if the waypoint and loiter_unlimited have the same position. We could prevent this reverse transversing if all the mission items in such a sequence would have the same PREVIOUS WAYPOINT assigned.

**Comment by julianoes** [comment]
@dayjaby I suppose for this case we should be using WAYPOINT, DELAY, IMAGE_START_CAPTURE, DELAY, but maybe that's not possible given the current implementation.

**Comment by coderkalyan** [comment]
If NAV_DELAY is supported in PX4, is there something preventing this from being implemented? Or can I take a crack at it (time permitting)?

**Comment by julianoes** [comment]
@coderkalyan feel free :)
